<h1>News Category Prediction</h6>

<b></b>
<h3>Importing required libraries:</h3>
    <p>The required libraries to solve the news category prediction algorithm are imported in the below section</p>

In [1]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.feature_extraction.text import CountVectorizer

<h3>Reading the data from dataset:</h3>
<p> In this section we read the data from them the <strong>News_Category_Dataset.json </strong>  file then we drop some fields and we convert the text data in the given dataset to the numeric format by using <strong> CountVectorizer</strong>.</p>

In [2]:
text=pd.read_json("News_Category_Dataset.json",lines=True)
text.drop(['link','date'], 1, inplace=True)
df=text
df=np.array(df)
tfidfvec = CountVectorizer(analyzer='word', ngram_range = (1,1),min_df = 1, max_features= 4)
tfidfdata = tfidfvec.fit_transform(df.ravel())

<h3>Train Test Split:</h3>
<p>In this section we read the features into <strong>X</strong> and target into <strong>y</strong> and then split the X and y into Train data and Test data.</p>

In [3]:
df=pd.DataFrame(tfidfdata.todense())
X = np.array(df.drop([1],1))
X=pd.DataFrame(X)
y = np.array(df[1])
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=42)

<h3>Logistic Regression:</h3>
<p>In this sectin we implement <strong>Logistic Regression</strong> algorithm by using the train data and test the score with test data.</p> 

In [7]:
clf = LogisticRegression(solver='lbfgs')
clf.fit(X_train,y_train)
y_pred=clf.predict(X_test)
y_pred=clf.predict_proba(X_test) 

print(clf.score(X_test, y_test))


0.875522771629108


<h3>Conclusion</h3>
     <p>Finally we are concluded the prediction with the score value of <strong> 0.876 </strong>.</p>